feat(container): add usage of new APIs (#8668)
withastro/docs: src/content/docs/en/reference/container-reference.mdx

```ts
export type AstroContainerOptions = {
	streaming?: boolean;
	renderers?: AddServerRenderer[];
};

export type AddServerRenderer =
	| {
			renderer: NamedSSRLoadedRendererValue;
			name: never;
	  }
	| {
			renderer: SSRLoadedRendererValue;
			name: string;
	  };
```

> #### `streaming` option

Reviewer: ```suggestion
### `streaming` option
```

> #### `renderers` option

Reviewer: ```suggestion
### `renderers` option
```

**Type:** `AddServerRenderer[]`

A list of loaded client renderers required by the component. Use this if your `.astro` component renders any [UI framework components](/en/guides/framework-components/) or MDX using an official Astro integration (e.g. React, Vue, etc.).

Renderers can be added through the Container API automatically for [static applications](/en/basics/rendering-modes/#pre-rendered), or cases where the container isn't called at runtime (e.g. testing with `vitest`).

For [on-demand rendered applications](/en/basics/rendering-modes/#on-demand-rendered), or cases where the container is called at runtime or inside other "shells" (e.g. PHP, Ruby, Java, etc.), renderers must be manually imported.

> ##### Adding a renderer through the Container API

Reviewer: ```suggestion
#### Adding a renderer through the Container API
```

const result = await container.renderToString(ReactWrapper);
```

> ##### Adding a renderer manually

Reviewer: ```suggestion
#### Adding a renderer manually
```

When the container is called at runtime, or inside other "shells", the `astro:container` virtual module's helper functions are not available. Import the necessary server renderer manually and store it inside the container using `experimental_Astro.Container.addServerRenderer`.

> In the following example, we manually import the server renderers for React and Vue components, and MDX for `.mdx` pages. 

Reviewer: ```suggestion
The following example manually imports the server renderers for React and Vue components, and MDX for `.mdx` pages. 
```

```js
import reactRenderer from "@astrojs/react/server.js";
import vueRenderer from "@astrojs/vue/server.js";
import mdxRenderer from "astro/jsx/server.js";

> const container = await experimental_AstroContainer.create();

Reviewer: @ematipico I wasn't sure whether we still also need the `experimental` prefix here and in the text in line 89? I added it because the other example does use it, but you can remove it if it's not correct.

Author: Oh yeah, that sounds good
